src/jit/gentree.cpp

```diff
@@ -13758,11 +13759,14 @@ void ReturnTypeDesc::Initialize(Compiler* comp, CORINFO_CLASS_HANDLE retClsHnd)
         }
     }
 
+#elif defined(_TARGET_X86) && !defined(LEGACY_BACKEND)
```

Comment:
Here we don't need second part of the condition. The way to look at is that ReturnTypeDesc implementation is for x86 and right now it is used by RyuJIT.